cmd/thv-operator/README.md

```diff
@@ -0,0 +1,224 @@
+# ToolHive Kubernetes Operator
+
+The ToolHive Kubernetes Operator manages MCP (Model Context Protocol) servers in Kubernetes clusters. It allows you to define MCP servers as Kubernetes resources and automates their deployment and management.
+
+This operator is built using [Kubebuilder](https://book.kubebuilder.io/), a framework for building Kubernetes APIs using Custom Resource Definitions (CRDs).
+
+## Overview
+
+The operator introduces a new Custom Resource Definition (CRD) called `MCPServer` that represents an MCP server in Kubernetes. When you create an `MCPServer` resource, the operator automatically:
+
+1. Creates a Deployment to run the MCP server
+2. Sets up a Service to expose the MCP server
+3. Configures the appropriate permissions and settings
+4. Manages the lifecycle of the MCP server
+
+## Installation
+
+### Prerequisites
+
+- Kubernetes cluster (v1.19+)
+- kubectl configured to communicate with your cluster
+
+### Installing the Operator
+
+1. Install the CRD:
+
+```bash
+kubectl apply -f deploy/operator/crds/toolhive.stacklok.dev_mcpservers.yaml
+```
+
+2. Create the operator namespace:
+
+```bash
+kubectl apply -f deploy/operator/namespace.yaml
+```
+
+3. Set up RBAC:
+
+```bash
+kubectl apply -f deploy/operator/rbac.yaml
+```
+
+4. Deploy the operator:
+
+```bash
+kubectl apply -f deploy/operator/operator.yaml
+```
+
+## Usage
+
+### Creating an MCP Server
+
+To create an MCP server, define an `MCPServer` resource and apply it to your cluster:
+
+```yaml
+apiVersion: toolhive.stacklok.dev/v1alpha1
+kind: MCPServer
+metadata:
+  name: fetch
+spec:
+  image: docker.io/mcp/fetch
+  transport: stdio
+  port: 8080
+  permissionProfile:
+    type: builtin
+    name: network
+  resources:
+    limits:
+      cpu: "100m"
+      memory: "128Mi"
+    requests:
+      cpu: "50m"
+      memory: "64Mi"
+```
+
+Apply this resource:
+
+```bash
+kubectl apply -f your-mcpserver.yaml
+```
+
+### Using Secrets
+
+For MCP servers that require authentication tokens or other secrets:
+
+```yaml
+apiVersion: toolhive.stacklok.dev/v1alpha1
+kind: MCPServer
+metadata:
+  name: github
+spec:
+  image: docker.io/mcp/github
+  transport: stdio
+  port: 8080
+  permissionProfile:
+    type: builtin
+    name: network
+  secrets:
+    - name: github-token
+      target: GITHUB_PERSONAL_ACCESS_TOKEN
```

Comment:
Should we make this `key`? Just to stay consistent with other [Kubernetes nomenclature](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables) around sourcing secrets.

Reply:
changed it, wdyt?